src/test/java/uk/gov/homeoffice/digital/sas/accruals/controllers/AccrualsControllerIntegrationTest.java

```diff
@@ -0,0 +1,252 @@
+      UUID.fromString("722875f4-e5de-40fa-b84d-eea99d1fba66");
+
+  private static final LocalDate TIME_ENTRY_START_DATE = LocalDate.of(
+      2023, 4, 2);
+
+  private static final LocalDate AGREEMENT_START_DATE = LocalDate.of(
+      2023,4,1);
+
+  private static final LocalDate AGREEMENT_END_DATE = LocalDate.of(
+      2023,  4,3);
+
+  private String agreementId;
+  @Autowired
+  MockMvc mvc;
+
+  @Autowired
+  AccrualsRepository accrualsRepository;
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    Agreement agreement = createAgreement(AGREEMENT_START_DATE, AGREEMENT_END_DATE, PERSON_ID);
+
+    MvcResult result = postAgreement(agreement)
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.items", not(empty())))
+        .andReturn();
+
+    agreementId = JsonPath.read(result.getResponse().getContentAsString(), "$.items[0].id");
+  }
+
+  @AfterEach
+  void tearDown() {
+    accrualsRepository.deleteAll();
+  }
+
+
+  @Test
+  void getAccrualsImpactedByTimeEntry_whenEndPointIsCalled_shouldReturnEmptyList()
+      throws Exception {
+     String timeEntryId = "4d254823-0a7d-43b4-b948-b43266c9cbc1";
+
+        getImpactedAccruals(timeEntryId)
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.items", empty()))
+        .andReturn();
+  }
+
+  @Test
+  void getAccrualsImpactedByTimeEntry_whenPriorAccrualIsPresent_shouldGetPriorAndAccrualWithContribution()
+      throws Exception {
+
+    setUpAccrualsData(LocalDate.of(2023, 4, 1), 1, agreementId);
+
+    String timeEntryId = getStoredTimeEntryId(LocalDate.of(2023,4,1));
+
+    getImpactedAccruals(timeEntryId)
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.items", not(empty())))
+        .andExpect(jsonPath("$.items.length()", is(2)))
+        .andExpect(jsonPath("$.items[0].accrualDate", is("2023-04-01")))
+        .andExpect((jsonPath("$.items[0].contributions.timeEntries", hasKey(timeEntryId))))
+        .andExpect(jsonPath("$.items[1].accrualDate", is("2023-04-02")))
+        .andReturn();
+  }
+
+  @Test
+  void getAccrualsImpactedByTimeEntry_whenEarlierContributionIsPresent_shouldGetContributionBeforeTimeEntry()
+      throws Exception {
+
+    setUpAccrualsData(LocalDate.of(2023, 3, 31), 2, agreementId);
+
+    String timeEntryId = getStoredTimeEntryId(LocalDate.of(2023,4,1));
+
+    getImpactedAccruals(timeEntryId)
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.items", not(empty())))
+        .andExpect(jsonPath("$.items.length()", is(3)))
+        .andExpect(jsonPath("$.items[0].accrualDate", is("2023-03-31")))
+        .andExpect(jsonPath("$.items[1].accrualDate", is("2023-04-01")))
+        .andExpect((jsonPath("$.items[1].contributions.timeEntries", hasKey(timeEntryId))))
+        .andExpect(jsonPath("$.items[2].accrualDate", is("2023-04-02")));
+  }
+
+  @Test
+  void getAccrualsImpactedByTimeEntry_whenTimeEntryIsBetweenCurrentAccrual_shouldGetContributionCoveringTimeEntry()
```

Comment:
Do we have tests that cover
`getAccrualsImpactedByTimeEntry_whenTimeEntryEndsInDifferentAgreement....`
This is when a time entry spans 2 agreements
`getAccrualsImpactedByTimeEntry_whenPreviousContributionIsInPreviousAgreement....`
When time entry previously covered last day of an agreement but was then updated to the following day
`getAccrualsImpactedByTimeEntry_whenPreviousContributionIsInFutureAgreement....`
When time entry previously covered first day of an agreement but was then updated to end the previous day

Reply:
getAccrualsImpactedByTimeEntry_whenTimeEntryEndsInDifferentAgreement....
https://github.com/UKHomeOffice/callisto-accruals-restapi/pull/28/commits/6ca3208b0a467c2e076b208ec5cabf487495c6f9#:~:text=getAccrualsImpactedByTimeEntry_timeEntryEndsInDifferentAgreement_returnEndOfSecondAgreement

getAccrualsImpactedByTimeEntry_whenPreviousContributionIsInPreviousAgreement....
https://github.com/UKHomeOffice/callisto-accruals-restapi/pull/28/commits/6ca3208b0a467c2e076b208ec5cabf487495c6f9#:~:text=getAccrualsImpactedByTimeEntry_previousContributionIsInPreviousAgreement_returnEndOfSecondAgreement

getAccrualsImpactedByTimeEntry_whenPreviousContributionIsInFutureAgreement...
https://github.com/UKHomeOffice/callisto-accruals-restapi/pull/28/commits/6ca3208b0a467c2e076b208ec5cabf487495c6f9#:~:text=getAccrualsImpactedByTimeEntry_previousContributionIsInFutureAgreement_returnEndOfFirstAgreement